parlai/core/torch_generator_agent.py

```diff
@@ -1359,4 +1362,6 @@ def select_paths(self, logprobs, prior_scores):
         # Convert back to logspace.
         scores = sprobs[hyp_ids, choices].log()
         best_scores = prior_scores.expand_as(scores) + scores
+        # In the case that any value of scores is very low, we may overflow
+        best_scores.clamp_min_(min=neginf(best_scores.dtype))
```

Comment:
Hm, I actually thought scores might be in fp16.

We’ll need this in topk too

Reply:
Search strategies are now guaranteed to be working with fp32, so this line is no longer needed.

To double-check, I switch to fp32 and verified that overflow is not an issue:
```
(Pdb) best_scores                                                              
tensor([-18.2475,  -5.9636,  -9.7469, -10.6112,  -9.9314, -15.8886,  -7.4628,
        -16.5711, -12.8132,  -6.3287,  -8.2394,  -8.3234, -16.6441,  -5.2373,                                                                                                                                                                
        -12.4562,  -6.0306,  -6.6018,  -5.3095,  -8.0521, -10.1184],        
       device='cuda:0')                                                                    
(Pdb) best_scores[0] = neginf(torch.float32)                                                                                                                                                                                                  
(Pdb) best_scores                                                     
tensor([-1.0000e+20, -5.9636e+00, -9.7469e+00, -1.0611e+01, -9.9314e+00,        
        -1.5889e+01, -7.4628e+00, -1.6571e+01, -1.2813e+01, -6.3287e+00,   
        -8.2394e+00, -8.3234e+00, -1.6644e+01, -5.2373e+00, -1.2456e+01, 
        -6.0306e+00, -6.6018e+00, -5.3095e+00, -8.0521e+00, -1.0118e+01],                
       device='cuda:0')                                                        
(Pdb) best_scores - 100                                                          
tensor([-1.0000e+20, -1.0596e+02, -1.0975e+02, -1.1061e+02, -1.0993e+02,  
        -1.1589e+02, -1.0746e+02, -1.1657e+02, -1.1281e+02, -1.0633e+02,               
        -1.0824e+02, -1.0832e+02, -1.1664e+02, -1.0524e+02, -1.1246e+02,  
        -1.0603e+02, -1.0660e+02, -1.0531e+02, -1.0805e+02, -1.1012e+02],                       
       device='cuda:0')
(Pdb) best_scores - 100000000000000
tensor([-1.0000e+20, -1.0000e+14, -1.0000e+14, -1.0000e+14, -1.0000e+14,
        -1.0000e+14, -1.0000e+14, -1.0000e+14, -1.0000e+14, -1.0000e+14,
        -1.0000e+14, -1.0000e+14, -1.0000e+14, -1.0000e+14, -1.0000e+14,
        -1.0000e+14, -1.0000e+14, -1.0000e+14, -1.0000e+14, -1.0000e+14],
       device='cuda:0')
```